README.md

```diff
@@ -11,47 +11,116 @@ ADaM in R Asset Library
+for all ADaM datasets outside of study specific needs. It depends on the user's collaboration
+and contribution to help grow over time to an asset library that is robust, easy to use and
+has an across-industry focus. We do not see a coverage of 100% of all ADaM derivations as ever
+achievable - ADaM is endless.
+
+We will provide:
+* A toolbox of re-usable functions and utilities to create ADaM datasets using R scripts in a
+modular manner (an "opinionated" design strategy)
+* Pharmaceutical communities and companies are encouraged to contribute to `{admiral}` following
+the provided programming strategy and modular approach
+* Functions that are comprehensively documented and tested, including example calls - these are
+all listed in the [Reference section](https://pharmaverse.github.io/admiral/reference/index.html)
+* Vignettes on how to create ADSL, BDS and OCCUR datasets, including example scripts
+* Vignettes for ADaM dataset specific functionality (i.e. dictionary coding, date imputation,
+SMQs ...)
+
+## Types of Packages
+There will be 3 foreseeable types of `{admiral}` packages:
+* Core package - one package containing all core functions required to create ADaMs,
+usable by any company (i.e. general derivations, utility functions and checks for ADSL, OCCDS and BDS)
+* TA (Therapeutic Area) package extensions - one package per TA with functions that are
+specific to algorithms and requirements for that particular TA (e.g. `{admiralonco}`)
+* Company package extensions - specific needs and plug-ins for the company, such as access to metadata
+(e.g. `{admiralroche}` or `{admiralgsk}`)
+
+## Admiral Manifesto
+For `{admiral}` and all extension packages, we prioritise providing our users with a **simple to adopt** toolkit
+that enables them to produce **readable** and **easily constructible** ADaM programs. The following explains
+our philosophy, which we try to adhere to across the `{admiral}` family of packages.
+There isn’t always a clear single, straight forward rule but there are guiding principles we adhere to.
+This manifesto helps show the considerations of our developers when making decisions.
+
+We have four priority design principles to achieve the main goal as follows:
+
+_**Usability**_
+
+All `{admiral}` functions should be easy to use. 
+* Documentation is an absolute priority. Each function reference page should cover the purpose,
+descriptions of each argument with permitted values, the expected input and output, with clear real-life
+examples - so that users don’t need to dig through code to find answers.
+* Vignettes that complement the functional documentation to help users see how best the functions can be
+applied to achieve customary ADaM requirements.
+* Functions should be written and structured in a way that users are able to read, re-use or extend them
+for study specific purposes if needed (see Readability below).
+
```

Comment:
Should we add some verbiage around how we prioritize Usability over Simplicity - while our goal is to make the functions easy to use, we don't want to make it so simple that nobody will actually use it.  There may be cases where adding a parameter or two will lead to much greater use of the function by end users.  So we are always debating that right balance, but in the end Usability takes priority over Simplicity.

Reply:
@upadhyor any specific text you would add or change for this point? i'm finding it difficult to see where to update. my view is the 2 kind of go hand in hand. for me the sentence below that says _"if you as a user pick up a function with >10 different arguments then chances are it is going to be difficult to understand if this function could be applied for your specific need"_ gives the general idea we're getting at. basically if there is a common need for users then we can add things to admiral functions but not at the cost of making them so complex that nobody will understand them. also we have this line in the programming strategy which i think speaks to your point about not over-simplifying: _"If something needed for ADaM could be achieved simply via an existing tidyverse function, then we do not wrap this into an admiral function, as that would add an unnecessary extra layer for users."_